### ISSUE TITLE
Found XDG-Music (uh no you didn't)

### ISSUE THREAD

**Issue by nPHYN1T3** [Taiko2k/Tauon#1240]
Sorry for the cheeky title but I was looking for a new player since I've never really been happy with the others I've used and I'd never seen this before. So I installed, added my music directory...and waited...and waited...and waited. It's been well over an hour and looking at the track count something is extremely wrong. So I killed it and started digging into what the hell is going on.

Running from a terminal it seems to be ignoring my Music directory and spits out XDG-Downloads and XDG-Music which are both wrong and are pointing to my whole home folder...which has over 20 years of DAW/Studio files on it.

So it's not only ignoring the user selected / defined Music folder but it's also got the XDG_DOWNLOAD_DIR and XDG_MUSIC_DIR environment variables wrong.

**Comment by Taiko2k** [comment]
Yep there's an issue in Tauon that it doesn't read the environment variable overrides for these. Ive committed a fix for this now.

Tauon does already read the settings from `~/.config/user-dirs.dirs`. Could you check what those are set to on your system?

**Comment by nPHYN1T3** [comment]
user-dirs.dirs is in fact wrong. I'm fairly sure the user-dirs.dirs file is ancient and has never been updated/configured correctly. (I've since deleted it)

```
XDG_DESKTOP_DIR="$HOME/Desktop"
XDG_DOWNLOAD_DIR="$HOME/"
XDG_TEMPLATES_DIR="$HOME/Templates"
XDG_PUBLICSHARE_DIR="$HOME/"
XDG_DOCUMENTS_DIR="$HOME/Documents"
XDG_MUSIC_DIR="$HOME/"
XDG_PICTURES_DIR="$HOME/"
XDG_VIDEOS_DIR="$HOME/"
```

However this is my environment

export XDG_STATE_HOME=$HOME/.local/state
export XDG_DATA_HOME=$HOME/.local/share
export XDG_CACHE_HOME=$HOME/.cache
export XDG_DATA_DIRS=/usr/local/share:/usr/share
export XDG_CONFIG_HOME=$HOME/.config
export XDG_CONFIG_DIRS=/etc/xdg/autostart
export XDG_DESKTOP_DIR="$HOME/Desktop"
export XDG_DOCUMENTS_DIR="$HOME/Documents"
export XDG_DOWNLOAD_DIR="$HOME/Downloads"
export XDG_MUSIC_DIR="$HOME/Music"
export XDG_PICTURES_DIR="$HOME/Pictures"
export XDG_PUBLICSHARE_DIR="$HOME/Public"
export XDG_TEMPLATES_DIR="$HOME/Templates"
export XDG_VIDEOS_DIR="$HOME/Videos"

**Comment by Taiko2k** [comment]
This article mentions how those are set https://wiki.archlinux.org/title/XDG_user_directories, but yeah next release of Tauon it should read these from the environment variables as first priority.

**Comment by nPHYN1T3** [comment]
Yeah there is no xdg-user-dirs update service on my machine. Looks like the only bits of that installed are because of Steam dependencies.

**Comment by C0rn3j** [comment]
@Taiko2k If you happen to know from the top of your head what's left to do here, could you describe it?

If not, don't bother, I'll have to go through the code myself anyway.

I'll implement this likely in the next couple of days, as I cornered myself in my [devel](https://github.com/Taiko2k/Tauon/compare/master...C0rn3j:Tauon:devel) branch and need working `XDG_DATA_HOME` at minimum to proceed.

Just dropping this here so I don't have to look it up again:  
https://wiki.archlinux.org/title/XDG_Base_Directory#User_directories  
https://specifications.freedesktop.org/basedir-spec/latest/

**Comment by nPHYN1T3** [comment]
Thanks for fixing this BTW.
